depmapomics/ccle.py

```diff
@@ -0,0 +1,1423 @@
+        mystorage_id (str, optional): google sheet storage id to access the ref sheets . Defaults to MYSTORAGE_ID.
+        sheetcreds (str, optional): @see updateTracker. Defaults to SHEETCREDS.
+        sheetname (str, optional): @see updateTracker. Defaults to SHEETNAME.
+        refsheet_url (str, optional): @see updateTracker. Defaults to REFSHEET_URL.
+        prevgenecn (str, optional): where the previous version exists on taiga (for QC purposes). Defaults to tc.get(name=TAIGA_ETERNAL, file='CCLE_gene_cn').
+        taiga_dataset (str, optional): where to save the output to on taiga. Defaults to TAIGA_CN.
+        dataset_description (str, optional): A long string that will be pushed to taiga to explain the CN dataset. Defaults to CNreadme.
+        subsetsegs (list[str], optional): what columns to keep for the segments. Defaults to [SAMPLEID, 'Chromosome', 'Start', 'End', 'Segment_Mean', 'Num_Probes', 'Status', 'Source'].
+        bamqc ([type], optional): @see updateTracker. Defaults to BAMQC.
+        procqc ([type], optional): @see updateTracker. Defaults to PROCQC.
+        source_rename ([type], optional): @see managing duplicates. Defaults to SOURCE_RENAME.
+    """
+    from taigapy import TaigaClient
+
+    tc = TaigaClient()
+
+    if prevgenecn == "ccle":
+        prevgenecn = tc.get(name=TAIGA_ETERNAL, file="CCLE_gene_cn")
+
+    tracker = pd.DataFrame()
+    if trackerobj is not None:
+        tracker = trackerobj.read_seq_table()
+
+    assert len(tracker) != 0, "broken source for sample tracker"
+
+    # doing wes
+    folder = os.path.join("output", samplesetname, "wes_")
+    if redoWES:
+        print("doing wes")
+        priority = tracker[
+            (tracker.datatype == "wes") & (tracker.prioritized == 1)
+        ].index.tolist()
+        todropwes = (
+            todrop
+            + tracker[
+                (tracker.datatype == "wes") & (tracker.blacklist == 1)
+            ].index.tolist()
+        )
+        (
+            wessegments,
+            genecn,
+            wesfailed,
+            wes_purecn_segments,
+            wes_purecn_genecn,
+            wes_loh,
+            wes_purecn_table,
+            _,
+        ) = cn.postProcess(
+            wesrefworkspace,
+            setEntity=wessetentity,
+            sampleset=AllSamplesetName if AllSamplesetName else samplesetname,
+            todrop=todropwes,
+            save_output=folder,
+            priority=priority,
+            genechangethresh=genechangethresh,
+            segmentsthresh=segmentsthresh,
+            maxYchrom=maxYchrom,
+            **kwargs,
+        )
+
+        # annotating source
+        for v in set(wessegments[SAMPLEID]):
+            wessegments.loc[
+                wessegments[wessegments[SAMPLEID] == v].index, "Source"
+            ] = tracker.loc[v, "source"]
+
+        wessegments.Source = wessegments.Source.replace(source_rename)
+        wessegments.Source += " WES"
```

Comment:
no need for it now. it is in the ID

Reply:
Yeah users can figure out which profile id maps to which data type with gumbo tables, but maybe it'd be good to keep the source here so it's more straightforward and they don't have to look it up?